controllers/cpms/cpms_controller_test.go

```diff
@@ -41,6 +44,41 @@ func makeTestCPMS(name, namespace string, cpmsSpec machinev1.ControlPlaneMachine
 	return cpms
 }
 
+func makeTestMachineSpecAWS() *runtime.RawExtension {
```

Comment:
I'd like to see two tests for checking what happens if the CPMS is not setting the `machineType` to `machines_v1beta1_machine_openshift_io` so we can see that this code is future-proof.

I'd also like to see how it handles that case if this operator is incorrectly running on a non-AWS non-GCP cloud provider.

Reply:
I added two tests 
- for an unsupported cloud provider
- for an invalid machine type

Please let me know if you think we should test anything else :)